### ISSUE TITLE
Support for Unity 5

### ISSUE THREAD

**Issue by DoctorJellyface** [Unity3D-Wine-Support/Unity3D-on-Wine#43]
Unity 5 is out (yay!), and among other things it introduces new ways to download and install Unity:

1. **Installer:** I haven't checked out this one, but it seems to be the easiest way - 636KB, downloads Unity automatically, so we don't need to update the script with a new SHA1SUM every update, not subject to change much - what's not to like? However, **I haven't tested this**, so I don't know what it installs along Unity.
2. **Direct downloads:** everything now has to be downloaded separately (yay?) - there's Unity+MD (64/32 bit), Standard Assets, Shaders, Example project. If we choose this path, we'll have to decide what to install and what not to.
3. **Torrent file:** probably not the best way, since it needs significant user interaction, and is much more complicated (the script exits if the files are not there).

So yeah, which one? Anyone who has tried the installer, _please_ comment.

**Comment by allkhor** [comment]
Hi, i test Unity5 on ubuntu 14.04 x64 with wine 1.7.37 Nvidia Gtx 260(340.58 proprietary driver).  Unity x64 work, but not install .Net Framework 4.0 (not support on x64). Unity x86 works great, Nvidia with --force-opengl bugs present. Asset manager now work! Need more testing.

**Comment by DoctorJellyface** [comment]
Try the `unity3d_minimal.verb` instead. It should work fine with the 64 bit editor.

**Comment by allkhor** [comment]
Mono develope not open without .Net Framework 4.0

**Comment by hlarsen** [comment]
i just used the Unity 5 installer under PlayOnLinux 32-bit WINE version 1.7.37 and everything installed without issue, but there isn't text in the editor menus (sound similar to #34). installing the PoL Tahoma package causes the text to appear.

i'm a first-time Unity user so i can't speak to much else, however loading an included test project (a sports car and some obstacles) and pressing play seems to work fine.

**Comment by DoctorJellyface** [comment]
@allkhor it does open, although the WINEPREFIX is wrongly set when it is opened through Unity. Refer to #40.

@hlarsen nice, thanks! Could you describe the process of installing Unity with the installer, step-by-step preferably?

**Comment by hlarsen** [comment]
Install steps for Unity 5 under PlayonLinux:

1) Install PlayOnLinux and download the official Unity installer
2) Open PlayOnLinux and go to Tools > Manage Wine Version and download v1.7.37 x32
3) At the main PlayOnLinux menu hit Install > Install a non-listed program
4) During the installer, choose 'Install a program in a new virtual drive', give it a name, then select all three 'before installation options' (Use another version of Wine, Configure Wine, Install some libraries)
5) Select Wine version 1.7.37, and 32-bit
6) When wineconfig opens change windows version to Windows 7 and hit ok
7) Check the 'POL_Install_tahoma' script when selecting PoL scripts
8) Browse to the Unity 5 installer, go through as normal, and you should be good to go once it's done

Notes:
1) Wine 1.7.38 is available at this point, however I haven't tested it.
2) setting version to Win7 might be optional, it's just something i always do when setting up a new prefix.

**Comment by zak-reynolds** [comment]
I also got Unity 5 (64 bit) working generally using the method @hlarsen used above on Linux Mint 17.1 x64. I used the full windows installer that you get through clicking the small "Additional Downloads/For Windows" dropdown on Unity's website, and I used Wine v1.7.39 x64 and installed "tahoma" under the Install Components tab in PlayOnLinux configuration (after forgetting to do it initially, the first time I ran Unity there was no text). I can also confirm that setting the version to Win7 is optional; I didn't notice any difference between that and WinXPSp3.

It runs well enough to use, I loaded up my current project and getting above 60fps and everything seems to be working, although I am getting some screen tearing. I tried to use the -force-opengl argument, but ran into the crash mentioned in issue #2 and haven't found a workaround yet.

**Comment by marcs77** [comment]
I got Unity 5.0.0f4 32bits using @hlarsen method in Ubuntu 14.04 with wine version 1.7.37 and wine windows version Win7 and it works except for some minor issues such as crashing when clicking the get started button, a little bit of UI lag, and some weird visual bugs when I resize the tabs (Project, Hierarchy...) but I didn't get screen tearing though.

**Comment by DoctorJellyface** [comment]
Thanks for the input everyone. @hlarsen I meant the **Unity Installer**, the one you mentioned in step 8, but never mind.

**Comment by DoctorJellyface** [comment]
Damn, we really need something like loomio.

Okay, so if everyone agrees ('specially @Pendrokar), tomorrow I'm going to update the scripts. Since Unity functions well without **any** windows DLLs installed, I'm going to update the scripts likewise. If someone is having problems without something, **please speak up**.

So, tomorrow, I'm going to fork this repo. In the PoL and winetricks script, I'm going to change the installer to the 600KB one. Also, I'll remove all the DLL installations, make the style more consistent and update the header - take out the useless information like last edit date and insert license stuff and clean the usefull stuff.

Then I'll make a pull request. If you can test the new scripts it'll be super cool of you. If you don't like anything there, please write before it gets merged.

Thanks.

**Comment by Pendrokar** [comment]
@DoctorJellyface go ahead, I won't be using Unity on Linux for some time, since I am experimenting with a game idea for Kinect. So I have to use Windows.

**Comment by DoctorJellyface** [comment]
What license? GPLv3?

**Comment by aswinthomas** [comment]
Does anyone have issues opening monodevelop? My unity instalaltion works well, but monodevelop crashes

**Comment by DoctorJellyface** [comment]
Tried right now, and works OK. However, I'm running in the default wineprefix, which is probably the problem. It's covered in #36.

However, there's a solution. Actually, a few of them:
1. Use a [native MonoDevelop install](http://wiki.unity3d.com/index.php/Running_Unity_on_Linux_through_Wine#MonoDevelop_.28Native.29). Probably the best option you've got.
2. If you really want to use the builtin MonoDevelop, you can use @Radivarig's [scripts](https://github.com/Unity3D-Wine-Support/Unity3D-on-Wine/blob/master/text-editors-MonoDevelop/unity3d_builtin_monodevelop.sh). This is also covered on the [wiki](http://wiki.unity3d.com/index.php/Running_Unity_on_Linux_through_Wine#MonoDevelop_.28Native.29).
3. Reinstall Unity in the default wineprefix (no scripts, just `winetricks tahoma corefonts` afterwards).

**Comment by Schroedingers-Cat** [comment]
Can we use the Debugger with the native MonoDevelop?

**Comment by DoctorJellyface** [comment]
No idea. I haven't tried getting it working. But highlighting and completion works.

**Comment by TheRealCodeCube** [comment]
I am trying to run Unity from a flash drive, but it crashes when I try to create/open a project:
LICENSE SYSTEM [20150421 17:03:05] No start/stop license dates set
LICENSE SYSTEM [20150421 17:03:05] Next license update check is after 2015-04-22T23:14:16
Built from '5.0/release' branch; Version is '5.0.1f1 (5a2e8fe35a68) revision 5910159'; Using compiler version '160040219'
BatchMode: 0, IsHumanControllingUs: 1, StartBugReporterOnCrash: 1, Is64bit: 1, IsPro: 0
Initialize mono
Mono path[0] = 'J:/Unity/Editor/Data/Managed'
Mono path[1] = 'J:/Unity/Editor/Data/Mono/lib/mono/2.0'
Mono path[2] = 'J:/Unity/Editor/Data/UnityScript'
Mono config path = 'J:/Unity/Editor/Data/Mono/etc'
Using monoOptions --debugger-agent=transport=dt_socket,embedding=1,defer=y
IsTimeToCheckForNewEditor: Update time 0 current 1429660985
C:/users/me/My Documents/2D space game test thingie woah that's a long name!
Crash!!!
Crash in winutils::ProcessInternalCrash(PEXCEPTION_POINTERS pExInfo)Rebuilding Library because the asset database could not be found!
 
(Filename: C:/buildslave/unity/build/Editor/Src/AssetPipeline/AssetInterface.cpp Line: 2027)

Unhandled exception: page fault on write access to 0x00000002 in 64-bit code (0x0000000004581395).
Register dump:
 rip:0000000004581395 rsp:000000001046e548 rbp:00000001406359d0 eflags:00010212 (  R- --  I   -A- - )
 rax:0000000000000002 rbx:0000000000000000 rcx:000000001046d390 rdx:000000014063854a
 rsi:0000000000000000 rdi:00000000011eb880  r8:000000001046e2a0  r9:ffffffffc0000005 r10:000000001046d860
 r11:000000001046d1c8 r12:0000000004581390 r13:0000000000000000 r14:000000001046f700 r15:000000000033e010
Stack dump:
0x000000001046e548:  00000001406359ff 00007fffff78c000
0x000000001046e558:  00000000011eb880 00007fffff78c000
0x000000001046e568:  00007f770057c7fc 00007fffff78c000
0x000000001046e578:  00007f77005b5fe3 00007fffff78c000
0x000000001046e588:  00007f77005aed2d 00007fffff78c000
0x000000001046e598:  00007f77005b5f87 ffffffffffffffff
0x000000001046e5a8:  000000001046e5e8 000000001046e5d0
0x000000001046e5b8:  00007fffff78c000 00000001406359d0
0x000000001046e5c8:  0000000004581390 ffffffffffffffff
0x000000001046e5d8:  00007f77005d4f30 000000007b84bec0
0x000000001046e5e8:  00007fffff78c000 bc89a2d1e7ecde56
0x000000001046e5f8:  0000000004581390 0000000000000000
Backtrace:
=>0 0x0000000004581395 (0x00000001406359d0)
0x0000000004581395: addb	%al,(%rax)
Modules:
Module	Address					Debug info	Name (155 modules)
PE	          340000-          36e000	Deferred        geometrytoolbox
PE	          370000-          616000	Deferred        freeimage
PE	          620000-          713000	Deferred        umbraoptimizer64
PE	          720000-          8fe000	Deferred        libpq
PE	         6700000-         69da000	Deferred        mono
PE	         7ce0000-         7ef8000	Deferred        nss3
PE	         8020000-         8027000	Deferred        mozalloc
PE	         8030000-         804c000	Deferred        mozglue
PE	         8050000-         8444000	Deferred        gkmedias
PE	         8450000-         a103000	Deferred        xul
ELF	        7a800000-        7abd6000	Deferred        opengl32<elf>
  \-PE	        7a850000-        7abd6000	\               opengl32
ELF	        7b800000-        7bc89000	Deferred        kernel32<elf>
  \-PE	        7b820000-        7bc89000	\               kernel32
ELF	        7be00000-        7c103000	Deferred        <wine-loader>
PE	       140000000-       1430f0000	Deferred        unity
PE	       180000000-       180040000	Deferred        unwrap
ELF	      3c7f000000-      3c7f3ba000	Dwarf           libwine.so.1
ELF	      3c7f800000-      3c7fae3000	Deferred        libkrb5.so.3
ELF	      3c7fc00000-      3c7fe0f000	Deferred        libkrb5support.so.0
ELF	      3c80000000-      3c803ed000	Deferred        libcrypto.so.10
ELF	      3c80400000-      3c8064d000	Deferred        libgssapi_krb5.so.2
ELF	      3c80800000-      3c80a33000	Deferred        libk5crypto.so.3
ELF	      3c80c00000-      3c80e6f000	Deferred        libssl.so.10
ELF	      3c82a00000-      3c82c7e000	Deferred        libfreebl3.so
ELF	      3c82e00000-      3c83036000	Deferred        libcrypt.so.1
ELF	      3c83200000-      3c8347e000	Deferred        libcups.so.2
ELF	      3c83e00000-      3c84070000	Deferred        libtspi.so.1
ELF	      3c84200000-      3c8451f000	Deferred        libgnutls.so.28
ELF	      3f2ba00000-      3f2bc24000	Deferred        ld-linux-x86-64.so.2
ELF	      3f2c200000-      3f2c41c000	Deferred        libpthread.so.0
ELF	      3f2c600000-      3f2c804000	Deferred        libdl.so.2
ELF	      3f2ca00000-      3f2cc16000	Deferred        libz.so.1
ELF	      3f2ce00000-      3f2d108000	Deferred        libm.so.6
ELF	      3f2d200000-      3f2d425000	Deferred        liblzma.so.5
ELF	      3f2d600000-      3f2d86d000	Deferred        libpcre.so.1
ELF	      3f2da00000-      3f2dc25000	Deferred        libselinux.so.1
ELF	      3f2de00000-      3f2e017000	Deferred        libgcc_s.so.1
ELF	      3f2e600000-      3f2e81a000	Deferred        libresolv.so.2
ELF	      3f2ee00000-      3f2f010000	Deferred        libbz2.so.1
ELF	      3f2f200000-      3f2f408000	Deferred        libffi.so.6
ELF	      3f30200000-      3f30422000	Deferred        libxcb.so.1
ELF	      3f30600000-      3f30833000	Deferred        libpng16.so.16
ELF	      3f30a00000-      3f30c04000	Deferred        libxau.so.6
ELF	      3f30e00000-      3f31143000	Deferred        libx11.so.6
ELF	      3f31200000-      3f3142a000	Deferred        libexpat.so.1
ELF	      3f31a00000-      3f31c12000	Deferred        libxext.so.6
ELF	      3f31e00000-      3f320aa000	Deferred        libfreetype.so.6
ELF	      3f32200000-      3f32449000	Deferred        libdbus-1.so.3
ELF	      3f32a00000-      3f32c3f000	Deferred        libfontconfig.so.1
ELF	      3f32e00000-      3f3300a000	Deferred        libxrender.so.1
ELF	      3f33200000-      3f33403000	Deferred        libxinerama.so.1
ELF	      3f33600000-      3f3380a000	Deferred        libxrandr.so.2
ELF	      3f33a00000-      3f33c06000	Deferred        libxfixes.so.3
ELF	      3f34600000-      3f34804000	Deferred        libcom_err.so.2
ELF	      3f34e00000-      3f35010000	Deferred        libxi.so.6
ELF	      3f35200000-      3f3540b000	Deferred        libxcursor.so.1
ELF	      3f35600000-      3f35806000	Deferred        libxxf86vm.so.1
ELF	      3f36600000-      3f36804000	Deferred        libkeyutils.so.1
ELF	      3f3d000000-      3f3d203000	Deferred        libxcomposite.so.1
ELF	      3f45a00000-      3f45c2a000	Deferred        libtinfo.so.5
ELF	      3f48400000-      3f48666000	Deferred        libp11-kit.so.0
ELF	      3f4a200000-      3f4a478000	Deferred        libgmp.so.10
ELF	      3f4b200000-      3f4b432000	Deferred        libnettle.so.4
ELF	      3f4be00000-      3f4c027000	Deferred        libhogweed.so.2
ELF	      3f4c200000-      3f4c428000	Deferred        libncurses.so.5
ELF	      3f4ca00000-      3f4cc13000	Deferred        libtasn1.so.6
ELF	      3f4e200000-      3f4e412000	Deferred        libavahi-client.so.3
ELF	      3f4fa00000-      3f4fc0d000	Deferred        libavahi-common.so.3
ELF	    7f76eeee3000-    7f76ef152000	Deferred        dbghelp<elf>
  \-PE	    7f76eeef0000-    7f76ef152000	\               dbghelp
ELF	    7f76ef152000-    7f76ef367000	Deferred        t2embed<elf>
  \-PE	    7f76ef160000-    7f76ef367000	\               t2embed
ELF	    7f76ef367000-    7f76ef580000	Deferred        userenv<elf>
  \-PE	    7f76ef370000-    7f76ef580000	\               userenv
ELF	    7f76ef580000-    7f76ef8b9000	Deferred        msvcp100<elf>
  \-PE	    7f76ef5c0000-    7f76ef8b9000	\               msvcp100
ELF	    7f76ef8b9000-    7f76efacd000	Deferred        msimg32<elf>
  \-PE	    7f76ef8c0000-    7f76efacd000	\               msimg32
ELF	    7f76efacd000-    7f76efd17000	Deferred        usp10<elf>
  \-PE	    7f76efae0000-    7f76efd17000	\               usp10
ELF	    7f76efd17000-    7f76f0000000	Deferred        msvcr100<elf>
  \-PE	    7f76efd40000-    7f76f0000000	\               msvcr100
ELF	    7f76f418c000-    7f76f4392000	Deferred        libnss_dns.so.2
ELF	    7f76f4392000-    7f76f4595000	Deferred        libnss_mdns4_minimal.so.2
ELF	    7f76f45c5000-    7f76f47da000	Deferred        mswsock<elf>
  \-PE	    7f76f45d0000-    7f76f47da000	\               mswsock
ELF	    7f76f47da000-    7f76f4b3b000	Deferred        wined3d<elf>
  \-PE	    7f76f47f0000-    7f76f4b3b000	\               wined3d
ELF	    7f76f4b3b000-    7f76f4d66000	Deferred        dxgi<elf>
  \-PE	    7f76f4b40000-    7f76f4d66000	\               dxgi
ELF	    7f76f4d66000-    7f76f4fa4000	Deferred        wbemprox<elf>
  \-PE	    7f76f4d70000-    7f76f4fa4000	\               wbemprox
ELF	    7f76f5143000-    7f76f537d000	Deferred        uxtheme<elf>
  \-PE	    7f76f5150000-    7f76f537d000	\               uxtheme
ELF	    7f76f537f000-    7f76f5623000	Deferred        winex11<elf>
  \-PE	    7f76f5390000-    7f76f5623000	\               winex11
ELF	    7f76f5685000-    7f76f58a2000	Deferred        jsproxy<elf>
  \-PE	    7f76f5690000-    7f76f58a2000	\               jsproxy
ELF	    7f76f58a2000-    7f76f5ae6000	Deferred        winhttp<elf>
  \-PE	    7f76f58b0000-    7f76f5ae6000	\               winhttp
ELF	    7f76f5ae6000-    7f76f5d64000	Deferred        setupapi<elf>
  \-PE	    7f76f5af0000-    7f76f5d64000	\               setupapi
ELF	    7f76f5d64000-    7f76f5f85000	Deferred        dnsapi<elf>
  \-PE	    7f76f5d70000-    7f76f5f85000	\               dnsapi
ELF	    7f76f5f85000-    7f76f61ad000	Deferred        imm32<elf>
  \-PE	    7f76f5f90000-    7f76f61ad000	\               imm32
ELF	    7f76f61ad000-    7f76f63da000	Deferred        msacm32<elf>
  \-PE	    7f76f61b0000-    7f76f63da000	\               msacm32
ELF	    7f76f63da000-    7f76f65ed000	Deferred        psapi<elf>
  \-PE	    7f76f63e0000-    7f76f65ed000	\               psapi
ELF	    7f76f65ed000-    7f76f697b000	Deferred        oleaut32<elf>
  \-PE	    7f76f6610000-    7f76f697b000	\               oleaut32
ELF	    7f76f697b000-    7f76f6bae000	Deferred        netapi32<elf>
  \-PE	    7f76f6980000-    7f76f6bae000	\               netapi32
ELF	    7f76f6bae000-    7f76f6de7000	Deferred        secur32<elf>
  \-PE	    7f76f6bc0000-    7f76f6de7000	\               secur32
ELF	    7f76f6de7000-    7f76f7012000	Deferred        iphlpapi<elf>
  \-PE	    7f76f6df0000-    7f76f7012000	\               iphlpapi
ELF	    7f76f7012000-    7f76f722f000	Deferred        wsock32<elf>
  \-PE	    7f76f7020000-    7f76f722f000	\               wsock32
ELF	    7f76f722f000-    7f76f74f2000	Deferred        winmm<elf>
  \-PE	    7f76f7240000-    7f76f74f2000	\               winmm
ELF	    7f76f74f2000-    7f76f77dc000	Deferred        crypt32<elf>
  \-PE	    7f76f7500000-    7f76f77dc000	\               crypt32
ELF	    7f76f77dc000-    7f76f7a25000	Deferred        winspool<elf>
  \-PE	    7f76f77e0000-    7f76f7a25000	\               winspool
ELF	    7f76f7a25000-    7f76f7d38000	Deferred        comctl32<elf>
  \-PE	    7f76f7a30000-    7f76f7d38000	\               comctl32
ELF	    7f76f7d38000-    7f76f7fcc000	Deferred        shlwapi<elf>
  \-PE	    7f76f7d50000-    7f76f7fcc000	\               shlwapi
ELF	    7f76f7fcc000-    7f76f843a000	Deferred        shell32<elf>
  \-PE	    7f76f7fe0000-    7f76f843a000	\               shell32
ELF	    7f76f843a000-    7f76f872e000	Deferred        comdlg32<elf>
  \-PE	    7f76f8440000-    7f76f872e000	\               comdlg32
ELF	    7f76f872e000-    7f76f8943000	Deferred        hid<elf>
  \-PE	    7f76f8730000-    7f76f8943000	\               hid
ELF	    7f76f8943000-    7f76f8b82000	Deferred        ws2_32<elf>
  \-PE	    7f76f8950000-    7f76f8b82000	\               ws2_32
ELF	    7f76f8b82000-    7f76f8e19000	Deferred        rpcrt4<elf>
  \-PE	    7f76f8b90000-    7f76f8e19000	\               rpcrt4
ELF	    7f76f8e19000-    7f76f91a5000	Deferred        ole32<elf>
  \-PE	    7f76f8e40000-    7f76f91a5000	\               ole32
ELF	    7f76f91a5000-    7f76f93bf000	Deferred        version<elf>
  \-PE	    7f76f91b0000-    7f76f93bf000	\               version
ELF	    7f76f93bf000-    7f76f9652000	Deferred        advapi32<elf>
  \-PE	    7f76f93d0000-    7f76f9652000	\               advapi32
ELF	    7f76f9652000-    7f76f99bb000	Deferred        gdi32<elf>
  \-PE	    7f76f9670000-    7f76f99bb000	\               gdi32
ELF	    7f76f99bb000-    7f76f9d9f000	Deferred        user32<elf>
  \-PE	    7f76f99e0000-    7f76f9d9f000	\               user32
ELF	    7f76f9d9f000-    7f76f9fac000	Deferred        libnss_files.so.2
ELF	    7f770051f000-    7f770082e000	Deferred        ntdll<elf>
  \-PE	    7f7700540000-    7f770082e000	\               ntdll
ELF	    7ffe7245f000-    7ffe72460000	Deferred        [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
0000000e services.exe
	0000001d    0
	0000001c    0
	00000016    0
	00000014    0
	00000010    0
	0000000f    0
00000012 winedevice.exe
	0000001b    0
	00000018    0
	00000017    0
	00000013    0
00000019 plugplay.exe
	0000001f    0
	0000001e    0
	0000001a    0
00000020 explorer.exe
	00000021    0
00000022 (D) J:\Unity\Editor\Unity.exe
	0000003d    0 <==
	0000003c   -1
	0000003b    0
	00000039    0
	00000038    0
	00000037    0
	00000036    0
	00000034    0
	00000033    0
	00000031    0
	00000030    0
	0000002f    0
	0000002e    0
	0000002d    0
	0000002c   -1
	0000002b    0
	0000002a    0
	00000029    0
	00000028    0
	00000026    0
	00000025    0
	00000024    0
	00000023    0

**Comment by aswinthomas** [comment]
Not sure if this is with unity 5 install only, but I'm getting this error when I run 'winetricks -q "./unity3d.verb"'

```
Note: command 'wine dotnetfx.exe /q /c:install.exe /q' returned status 26.  Aborting.
```

minimal version. works fine. With minimal you can install 64 bit and 32 bit Unity, but you get a black screen on Edit->preferences

I'd really like to use winetricks to make use of the wonderful native monodevelop scripts in this repo

**Comment by TheRealCodeCube** [comment]
I found a fix for my problem. I forget the exact path, but I needed to create an AppData folder in my wine directory.

**Comment by DoctorJellyface** [comment]
@TheRealCodeCube @aswinthomas what Wine version are you running?

Also, could anyone test the scripts from #47 please? They should be completely compatible with Unity 5, but no-one tested them yet. Thanks!

**Comment by TheRealCodeCube** [comment]
@DoctorJellyface wine-1.7.38 (Staging) on Fedora fc21

**Comment by Schroedingers-Cat** [comment]
I'm testing #47. Script fails to download tahoma32.exe. I remember this not working months ago. Did you test it?

**Comment by speculees** [comment]
Ok, guys. :) I installed Unity 5 (32 bit) on Arch Linux via wine (also 32 bit). No problems during installation. However, on starting the example project Unity crashes leaving me with an error in terminal:

"err:wgl:has_opengl Failed to load libGL: libnvidia-glcore.so.340.76: cannot open shared object file: No such file or directory
err:wgl:has_opengl OpenGL support is disabled."

...so after typing:

"$ ls /usr/lib32/ | grep nvidia
libnvidia-cfg.so
libnvidia-cfg.so.1
libnvidia-cfg.so.340.76
libnvidia-eglcore.so
libnvidia-glcore.so
libnvidia-glsi.so
libnvidia-tls.so
libnvidia-tls.so.340.76
nvidia"

...i did:

"ln -s /usr/lib/libnvidia-glcore.so.340.76 /usr/lib32/libnvidia-glcore.so.340.76"

...after trying to run it again I get this:

"err:wgl:has_opengl Failed to load libGL: libnvidia-glcore.so.340.76: wrong ELF class: ELFCLASS64
err:wgl:has_opengl OpenGL support is disabled."

Any ideas maybe? Tnx anyway ;-)

$ wine --version
wine-1.7.42

Nvidia 8800GT

**Comment by DoctorJellyface** [comment]
@TheRealCodeCube hm, that's strange, 1.7.38 should already have a fix for that problem.

@Schroedingers-Cat you're running Ubuntu, aren't you?

@speculees hm, nvidia again. Well, I'm not sure you can use a 64 bit library as a 32 bit one. Have you installed the 32 bit nvidia drivers? Try consulting the Arch wiki for that.

Also, why don't you guys just use the 64 bit version if you have 64 bit systems? It runs flawlessly, what do you need more?

**Comment by speculees** [comment]
I'll try that, tnx dude. I'll get back to you. :)

**Comment by Schroedingers-Cat** [comment]
@DoctorJellyface yes, it is Ubuntu Gnome 15.04. Is this a known issue?

**Comment by DoctorJellyface** [comment]
@Schroedingers-Cat not in 15.04, no. 14.04 had an issue with downloading Tahoma because of a faulty winetricks it had in the repos. Try downloading the newest winetricks from https://github.com/Winetricks/winetricks

@speculees happy to help.

**Comment by Schroedingers-Cat** [comment]
Using latest winetricks, I can start Unity 5 (x64). However, using the built-in MonoDevelop, I'm unable to start the debugging "Attach to process" does not show any process. Did this use to work?

**Comment by speculees** [comment]
Okay, I got it working. Tnx DoctorJellyface. :+1:

**Comment by DoctorJellyface** [comment]
@Schroedingers-Cat I don't know, I never tried.

@speculees great!

**Comment by paxamit** [comment]
I installed Unity 5.01 x64 on clean wine prefix. Arch x64, wine with opengl patch.
There was some errors from .NET during installation but every thing is working just fine. 
None custom setup required beside `winetricks tahoma`.

**Comment by speculees** [comment]
Ok. I have a minor problem now. :-) The thing is I can start MonoDevelop truogh Unity only once. If I close it I have to restart comuter before I can open it again...

**Comment by DoctorJellyface** [comment]
@paxamit do you experience the #10 inner window delay bug? Which patch did you use? PlayOnLinux only has 32 bit opengl patched wine afaik.

@speculees builtin or native? Did you try looking at the processes?

**Comment by DoctorJellyface** [comment]
@paxamit do you experience the #10 inner window delay bug? Which patch did you use? PlayOnLinux only has 32 bit opengl patched wine afaik.

@speculees builtin or native? Did you try looking at the processes?

**Comment by speculees** [comment]
I'm using builtin Unity Mono. On "ps aux | grep mono" i find this process: 
"1201  0.9  0.0      0     0 ?        Zsl  17:23   0:35 [MonoDevelop.exe] <defunct>"
Sometimes when I close Mono, this process (1201) gets killed, and afterwards there is no way to open it again without restarting OS. Restarting only Unity Does not help.

**Comment by paxamit** [comment]
unfortunately I experience bug #10 
but from quick tests performance without -force-opengl is even better so this isn't much of an issue for me.
I used pkg from here https://aur.archlinux.org/packages/wine-unity3d-git/ which use patches from https://github.com/Unity3D-Wine-Support/Unity3D-on-Wine.git (my wine-1.7.40-79)

**Comment by ZeronSix** [comment]
Is there a way to fix --force-opengl for 64-bit Unity3D installation?

**Comment by DoctorJellyface** [comment]
See issue #2.

**Comment by YugoCode** [comment]
I installed Unity with PlayOnLinux and Unity is working fine. I want to use the native monodevelop, so I installed it and followed these steps: http://wiki.unity3d.com/index.php/Running_Unity_on_Linux_through_Wine#MonoDevelop_.28Native.29

But still monodevelop is not opening. Can you help me?

**Comment by Radivarig** [comment]
@YugoCode are you trying to open .js files?

**Comment by anddam** [comment]
I'm experiencing the same issue running wine 1.7.29 under Debian.

I used winecfg to setup a prefix in ~/.local/share/wineprefixes/unity3d, manually installed mono using the wine-mono msi provided by winehq.org (my wine instance doesn't do it automatically even if told so at the initial dialog). Then I used [DoctorJellyface's latest winetricks verb](https://github.com/DoctorJellyface/Unity3D-on-Wine/blob/master/unity3d.verb) calling '''winetricks unity3d.verb'''.

Once installed I get the same output that [this thread shows](http://answers.unity3d.com/questions/724427/when-i-launch-uinty-there-is-no-text-please-help.html)

I tried to do the checks I could figure on my side, winetricks list-installed command confirms I have the following packages installed:
  -  corefonts
  -  ie8
  -  msls31
  -  tahoma
  -  vcrun2010

and I can see tahoma.ttf among the fonts in  .local/share/wineprefixes/unity3d/drive_c/windows/Fonts/
```
Arialbd.TTF  courbd.ttf    Georgiaz.TTF  Times.TTF     Verdana.TTF
Arialbi.TTF  courbi.ttf    Impact.TTF    Trebucbd.ttf  Verdanaz.TTF
Ariali.TTF   couri.ttf     tahomabd.ttf  trebucbi.ttf  Webdings.TTF
Arial.TTF    cour.ttf      tahoma.ttf    trebucit.ttf
AriBlk.TTF   Georgiab.TTF  Timesbd.TTF   trebuc.ttf
Comicbd.TTF  Georgiai.TTF  Timesbi.TTF   Verdanab.TTF
Comic.TTF    Georgia.TTF   Timesi.TTF    Verdanai.TTF
```

so I'm not sure if I should edit anything in registry at all as per #18 since tahoma is already there.

Is there a way to debug what the issue actually is?

I do not want to go through PlayOnLinux, I'm staying with system's wine/winetricks.

**Comment by YugoCode** [comment]
@Radivarig No, just csharp files. :(

**Comment by Radivarig** [comment]
Please replace native script with the following file as external editor, (give it +x rights) 
```
#!/bin/sh
echo $@ >> ~/u3d_args
```
open any file and post the content of u3d_args

**Comment by DoctorJellyface** [comment]
@anddam what issue are you referencing? It's not clear from your comment.

Right now I would suggest you to updated your Wine. The script installed IE8, which means that your wine is version is < 1.7.34, so for example your Asset Store won't work.

Also, you will want to try using a newer Winetricks.

Could you attach a screenshot so we  can see the problem? Thanks.

**Comment by anddam** [comment]
@DoctorJellyface my bad, I initially thought this ticket was specific to missing fonts like in #18 and #34 with the difference that this was for 5 release.

In my description I linked a message to answer.unity3d.com showing the very same issue, the direct link to the image in the initial message of that thread [is this](http://answers.unity3d.com/storage/attachments/27555-dsfdfsdfsdf.png), I get the very same output.
Notice that the missing font issue happens on a windows system in that thread.

Do you confirm you have a working Unity5 setup on your linux system?
There are many reference that this is due to missing fonts but I don't seem to be missing any of those, can you provide a list of the installed fonts in your wineprefix?

**Comment by DoctorJellyface** [comment]
Ah, I missed that, sorry.

Yes, I do have a working Unity 5 installation. There have been some references, but usually it was the fault of either old packages or incorrect installation. I would suggest you start over, and installing unity with just the winetricks script and see what it does. I don't think overwriting an existing prefix with winetricks is a good idea. What system are you running btw?

```
$ ls .wine/drive_c/windows/Fonts/
Arialbd.TTF  courbd.ttf    Georgiaz.TTF  Times.TTF     Verdana.TTF
Arialbi.TTF  courbi.ttf    Impact.TTF	 Trebucbd.ttf  Verdanaz.TTF
Ariali.TTF   couri.ttf	   tahomabd.ttf  trebucbi.ttf  Webdings.TTF
Arial.TTF    cour.ttf	   tahoma.ttf	 trebucit.ttf
AriBlk.TTF   Georgiab.TTF  Timesbd.TTF	 trebuc.ttf
Comicbd.TTF  Georgiai.TTF  Timesbi.TTF	 Verdanab.TTF
Comic.TTF    Georgia.TTF   Timesi.TTF	 Verdanai.TTF
```

**Comment by DoctorJellyface** [comment]
Also, PlayOnLinux users, please test #50. If it's correct, I'll merge and close this issue. Thanks.
-----------------------------------------------------------------------------------------------------------------------

**Comment by ElijahLynn** [comment]
http://blogs.unity3d.com/2015/07/01/the-state-of-unity-on-linux

**Comment by DoctorJellyface** [comment]
I'll merge #50 and close this issue.

**Comment by scoutroden** [comment]
Is anyone maintaining the script at https://github.com/Unity3D-Wine-Support/Unity3D-on-Wine/blob/master/unity.pol ?

The script to install Unity3d game development environment/tools has some bugs related to target download file version and the md5 hash not matching.


The current path to update the script with is:
`POL_Download "http://netstorage.unity3d.com/unity/0b02744d4013/UnityDownloadAssistant-5.0.2f1.exe" "dfdb5394ee22260da4aea09e5f233314"`
 
The current path to update the script with is:
http://netstorage.unity3d.com/unity/910d71450a97/UnityDownloadAssistant-5.3.3f1.exe
and the hash should be corrected to "b80a457c11dc30074d43dd3adefecab1"

I did manually update a copy of the script here, but I am not a pro like y'all, so when I changed the line to read for the new hash, it just ignored that, and still threw the same error, as if I had never changed it.
My new line is: 

`POL_Download "http://netstorage.unity3d.com/unity/910d71450a97/UnityDownloadAssistant-5.3.3f1.exe" "b80a457c11dc30074d43dd3adefecab1"`


Why didn't this fix the snag? How do I proceed to install Unity3d? Thanks!
